src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -326,8 +278,55 @@ class World internal constructor(
     /**
      * Returns the world's systems.
      */
-    var systems = emptyArray<IntervalSystem>()
-        internal set
+    private var _systems = mutableListOf<IntervalSystem>()
```

Comment:
I cannot remember anymore but I think mutableListOf creates a LinkedHashList or something. I don't know if the iteration over systems is then with an iterator or not.

We need to check that because otherwise we create a lot of iterator instances and trigger the garbage collector.

If possible, we should maybe just use an ArrayList here. I think that has the least overhead and iteration should be a simple one without iterator.